packages/vite/src/node/plugins/define.ts

```diff
@@ -31,16 +31,24 @@ export function definePlugin(config: ResolvedConfig): Plugin {
     })
   }
 
-  const env = { ...config.env }
   const userDefine: Record<string, string> = {}
+  const userDefineEnv: Record<string, string> = {}
   for (const key in config.define) {
     const val = config.define[key]
     userDefine[key] = typeof val === 'string' ? val : JSON.stringify(val)
 
     // make sure `import.meta.env` object has user define properties
-    const match = key.match(metaEnvRe)
-    if (match) {
-      env[match[1]] = val
+    if (isBuild) {
+      const match = key.match(metaEnvRe)
+      if (match) {
+        userDefineEnv[match[1]] =
+          // test if value is raw identifier to wrap with __vite__ so when
+          // stringified for `import.meta.env`, we can remove the quotes and
+          // retain being an identifier
+          typeof val === 'string' && /^[\p{L}_$]/u.test(val.trim())
```

Comment:
Ah  I see. String values will be explicitly quoted. So we can tell raw identifier via a head alphabet